system/Test/bootstrap.php

```diff
@@ -7,6 +7,14 @@
 // Make sure it recognizes that we're testing.
 $_SERVER['CI_ENVIRONMENT'] = 'testing';
 define('ENVIRONMENT', 'testing');
+defined('CI_DEBUG') || define('CI_DEBUG', true);
+
+// Often these constants are pre-defined, but query the current directory structure as a fallback
+defined('HOMEPATH') || define('HOMEPATH', realpath(rtrim(getcwd(), '\\/ ')) . DIRECTORY_SEPARATOR);
+$source = is_dir(HOMEPATH . 'app') ? HOMEPATH : 'vendor/codeigniter4/codeigniter4/';
```

Comment:
should be `vendor/codeigniter4/framework` instead for release path, not development path:

```suggestion
$source = is_dir(HOMEPATH . 'app') ? HOMEPATH : 'vendor/codeigniter4/framework/';
```

Need note addition, if using next version ( 4.0.5+), require `--prefer-source` for testing purpose if rely to `vendor/codeigniter4/framework/` directory.

Reply:
or better like this:

```php
$source = is_dir(HOMEPATH . 'app') 
       ? HOMEPATH 
       : (is_dir('vendor/codeigniter4/framework/') 
               ? 'vendor/codeigniter4/framework/' 
               : 'vendor/codeigniter4/codeigniter4/');
```

Reply:
For `--prefer-source` on testing purpose reason is on PR https://github.com/codeigniter4/CodeIgniter4/pull/3438

Reply:
I forgot it could be the release version, good catch. I like that second solution, covers all bases.